Lib/fontTools/merge.py

```diff
@@ -1067,26 +1067,23 @@ def merge(self, fontfiles):
 		# Take first input file sfntVersion
 		sfntVersion = fonts[0].sfntVersion
 
-		cffTables = []
+		cffTables = [None] * len(fonts)
 		if sfntVersion == "OTTO":
 			for i, font in enumerate(fonts):
 				font['CFF '].cff.desubroutinize()
-				cffTables.append(font['CFF '])
+				cffTables[i] = font['CFF ']
 
 		# Reload fonts and set new glyph names on them.
 		# TODO Is it necessary to reload font?  I think it is.  At least
 		# it's safer, in case tables were loaded to provide glyph names.
 		fonts = [ttLib.TTFont(fontfile) for fontfile in fontfiles]
-
-		if sfntVersion == "OTTO":
-			for font, glyphOrder, cffTable in zip(fonts, glyphOrders, cffTables):
-				font.setGlyphOrder(glyphOrder)
+		for font, glyphOrder, cffTable in zip(fonts, glyphOrders, cffTables):
+			font.setGlyphOrder(glyphOrder)
+			if cffTable:
 				# Rename CFF CharStrings to match the new glyphOrder.
+				# Using cffTable before reloading the fonts, because reasons.
```

Comment:
> Using cffTable before reloading the fonts, because reasons.

a bit cryptic, perhaps intentially so. But as far as I can tell, you are actually cffTable _after_ reloading, so I'm even more confused by the comment.

Reply:
ah I now see. You are using the cffTable from the original fonts, before they have been reloaded. Ok, then why?

Reply:
Yes. I don’t fully understand what is going on, but if we don’t do this then duplicated glyphs gets messed up (they seem to be using the outlines from the first font) in the output font. This seems to be related to the font reloading step, if we stop doing this then all this becomes moot. My testing does not show any other difference with or without reloading.